src/AvaloniaEdit/Search/TextBoxWithLabel.xaml.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Presenters;
+using Avalonia.Controls.Primitives;
+using Avalonia.Data;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.Markup.Xaml;
+using Avalonia.Media;
+using Avalonia.Metadata;
+using Avalonia.Threading;
+using System;
+using System.ComponentModel;
+using System.Reflection;
+
+namespace AvaloniaEdit.Search
+{
+    public class TextBoxWithLabel : TextBox
```

Comment:
Avalonia has this functionality built in to TextBox, just use the `Watermark` property.

Reply:
Overlooked that. Now using Watermark instead